web/src/main/java/org/springframework/security/web/authentication/rememberme/TokenBasedRememberMeServices.java

```diff
@@ -83,64 +95,86 @@
  */
 public class TokenBasedRememberMeServices extends AbstractRememberMeServices {
 
+	private static final RememberMeHashingAlgorithm DEFAULT_HASHING_ALGORITHM = RememberMeHashingAlgorithm.MD5;
+
+	private RememberMeHashingAlgorithm hashingAlgorithm = DEFAULT_HASHING_ALGORITHM;
+
 	public TokenBasedRememberMeServices(String key, UserDetailsService userDetailsService) {
 		super(key, userDetailsService);
 	}
 
+	/**
+	 * Constructs a {@link TokenBasedRememberMeServices} using the provided parameters
+	 * @param key the key to use in the hashing algorithm
+	 * @param userDetailsService the {@link UserDetailsService} to retrieve users from
+	 * @param hashingAlgorithm the hashing algorithm to use in the Cookie value
+	 */
+	public TokenBasedRememberMeServices(String key, UserDetailsService userDetailsService,
+			RememberMeHashingAlgorithm hashingAlgorithm) {
```

Comment:
I think that since this is an optional parameter, it would be ideal as a setter.